### PR TITLE
qiskit.test.mock is deprecated

### PR DESCRIPTION
**Pull request by WhiteSymmetry**
DeprecationWarning: The module 'qiskit.test.mock' is deprecated since Qiskit Terra 0.21.0, and will be removed 3 months or more later. Instead, you should import the desired object directly 'qiskit.providers.fake_provider'.
#from qiskit.test.mock import FakeVigo --> 
from qiskit.providers.fake_provider import FakeVigo
(https://qiskit.org/documentation/apidoc/providers_fake_provider.html)

### Summary
DeprecationWarning: The module 'qiskit.test.mock'

### Details and comments
DeprecationWarning: The module 'qiskit.test.mock' is deprecated since Qiskit Terra 0.21.0, and will be removed 3 months or more later. Instead, you should import the desired object directly 'qiskit.providers.fake_provider'.
#from qiskit.test.mock import FakeVigo -->
from qiskit.providers.fake_provider import FakeVigo
(https://qiskit.org/documentation/apidoc/providers_fake_provider.html)

### CODE REVIEW

File: tutorials/circuits_advanced/01_advanced_circuits.ipynb [hunk 1]
```diff
@@ -472,7 +472,8 @@
     "import time\n",
     "from itertools import combinations\n",
     "from qiskit.compiler import assemble\n",
-    "from qiskit.test.mock import FakeVigo\n",
+    "#from qiskit.test.mock import FakeVigo\n",
+    "from qiskit.providers.fake_provider import FakeVigo\n",
```
Comment (HuangJunye) [inline tutorials/circuits_advanced/01_advanced_circuits.ipynb:476]:
```suggestion
    "from qiskit.providers.fake_provider import FakeVigo\n",
```

Comment (HuangJunye) [reply tutorials/circuits_advanced/01_advanced_circuits.ipynb:476]: We can remove the old code.

**Review by HuangJunye — COMMENTED** [review]
Thanks a lot for your contribution!

**Review by Eric-Arellano — APPROVED** [review]
Thank you for making Qiskit better! 🎉

**Pull request merged**